### DISCUSSION TITLE
zfs extension with secureboot unsupported?

### DISCUSSION THREAD

**Discussion by roosmaa** [siderolabs/talos#9367] [Q&A] [answered]
I'm trying to deploy a baremetal (uefi/secureboot) node with zfs extension with the images from talos-factory, but the kernel module won't load automatically and when trying to modprobe manually I'm facing the `Operation not permitted` error.

The zfs.ko seems to be signed with some key (looking at the end of the file), but not sure if it's the correct one. There are no underlying error messages in dmesg either.

Has anyone managed to get zfs working with secureboot?

**Comment by smira** [comment, marked as answer]
You can't load modules manually, as this is prohibited by Talos Linux security model.

The proper way is to use machine configuration: https://www.talos.dev/v1.8/reference/configuration/v1alpha1/config/#Config.machine.kernel

**Reply by roosmaa** [reply]
Ahh, so that's what I was missing! Thanks 🙇

(I've got so used to managing the kernel related things via the talos-factory config, I forgot to check the machine config for hints 🤦)
